src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -254,14 +254,35 @@ private async Task RunChangeConsumptionLoopAsync()
                     await connection.OpenAsync(token);
                     this._logger.LogDebugWithThreadId("END OpenChangeConsumptionConnection");
 
+                    bool forceReconnect = false;
                     // Check for cancellation request only after a cycle of checking and processing of changes completes.
                     while (!token.IsCancellationRequested)
                     {
+                        bool isConnected = await connection.TryEnsureConnected(forceReconnect, this._logger, "ChangeConsumptionConnection", token);
+                        if (!isConnected)
+                        {
+                            // If we couldn't reconnect then wait our delay and try again
+                            await Task.Delay(TimeSpan.FromMilliseconds(this._pollingIntervalInMs), token);
+                            continue;
+                        }
+                        else
+                        {
+                            forceReconnect = false;
+                        }
                         this._logger.LogDebugWithThreadId($"BEGIN CheckingForChanges State={this._state}");
                         if (this._state == State.CheckingForChanges)
                         {
-                            await this.GetTableChangesAsync(connection, token);
-                            await this.ProcessTableChangesAsync(connection, token);
+                            try
+                            {
+                                await this.GetTableChangesAsync(connection, token);
+                                await this.ProcessTableChangesAsync(connection, token);
```

Comment:
If connection drops before executing 'ProcessTableChangesAsync()` - you'll be in ProcessingChanges state, so this will not repeat?

Reply:
Good catch! That's actually a bug with the current implementation. I'll file an issue and fix it as a follow up PR.

Reply:
https://github.com/Azure/azure-functions-sql-extension/issues/600